onadata/apps/api/tests/viewsets/test_metadata_viewset.py

```diff
@@ -192,6 +192,18 @@ def test_delete_metadata(self):
             self.assertEqual(response.status_code, 204)
             self.assertEqual(count, MetaData.objects.count())
 
+    def test_delete_xform_deletes_media_metadata(self):
+        for data_type in ['supporting_doc', 'media', 'source']:
+            self._add_form_metadata(self.xform, data_type,
+                                    self.data_value, self.path)
+        # Soft delete xform
+        self.xform.soft_delete()
+        # Confirm that media files were deleted
+        form_metadata = [
+            mt.data_type for mt in self.xform.metadata_set.filter(
+                deleted_at__isnull=True)]
+        self.assertNotIn('media', form_metadata)
```

Comment:
Shouldn't we be checking that the response from the viewset in this case is empty ?